### PR TITLE
add : single product page

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
### In this PR, I have added the following functionalities:
1. Show individual Product Details page on clicking the product
2. Add to cart and wishlist functionality on Individual Product page.
3. On home page, after clicking specific category navigate to products page with that filter.
4. Footer categories navigate to a specific category on the products page

### Files to check:
- ProductPage.jsx
- HomePage.jsx modifications

### Files to  ignore:
- CSS files
- Minor modifications
- database -> product.js

Please review this PR and give your suggestions

### [Live Link](https://dazzle-cerashop-git-single-product-page-apurvasawant11.vercel.app/)

### CODE REVIEW

File: src/pages/product/ProductPage.jsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  const wishlistHandler = () => {
+    token
+      ? isInWishlist
+        ? removeFromWishlist(product?._id, wishlistDispatch, token)
+        : addToWishlist(wishlistDispatch, product, token)
+      : navigate("/login");
+  };
+  return (
+    <section className="flex-row-center items-start wrap mt-2p5 product-section">
+      <img src={product?.imgURL} />
```
Comment (foolhardy21) [inline src/pages/product/ProductPage.jsx]: suggestion - you may add `alt` too in product.

**Review by foolhardy21 — APPROVED** [review]
clean code. good logic for `cart` and `wishlist` handlers.

File: src/components/footer/Footer.jsx [hunk 2]
```diff
@@ -61,7 +77,7 @@ const Footer = () => {
           Made with <i className="fas fa-heart"></i> by Apurva Sawant
```
Comment (ShekharDhangar) [inline src/components/footer/Footer.jsx:77]: You can use react icons, they are much better and in that you get all icons of different sites in one.

**Review by ShekharDhangar — APPROVED** [review]
Apart from react-icons suggestion, everything looks good. One thing you can do it that you can get users previous cart and wishlist data when user logged in again from api. It will go away on refresh but it will stay if user just logs in and logsout

File: src/components/footer/Footer.jsx (hunk 2 above)
Comment (ApurvaSawant11) [reply src/components/footer/Footer.jsx:77]: I tried react icons, but the size of each icon was differing. I'll try to add those when I refactor my code

**Comment by ApurvaSawant11** [comment]
> Apart from react-icons suggestion, everything looks good. One thing you can do it that you can get users previous cart and wishlist data when user logged in again from api. It will go away on refresh but it will stay if user just logs in and logsout

Right now I am clearing the wishlist and cart when the user logs out. But I'll try what you have suggested. Thanks!

File: src/pages/product/ProductPage.jsx (hunk 1 above)
Comment (ApurvaSawant11) [reply src/pages/product/ProductPage.jsx]: Yess. Thanks for pointing it out. Will update it

**Pull request merged**